skimage/meson.build

```diff
@@ -29,37 +29,51 @@ if is_windows
   endif
 endif
 
-# NumPy include directory - needed in all submodules
-incdir_numpy = run_command(py3,
-  [
-    '-c',
-    'import os; os.chdir(".."); import numpy; print(numpy.get_include())'
-  ],
-  check: true
-).stdout().strip()
+# Both NumPy and Pythran require header files that may differ between the build
+# system and the host system in a cross-compilation environment. To accommodate
+# these cases, we can query user-defined properties that can be specified in
+# the 'properties' section of a Meson cross file:
+#
+#   [properties]
+#   numpy-include-dir = '/path/to/numpy/includes'
+#   pythran-include-dir = '/path/to/pythran/includes'
+#
+# In the absence of explicitly configured paths, just run the build Python and
+# try to query the Python packages for their paths directly.
+
+# NumPy include directory
+incdir_numpy = meson.get_external_property('numpy-include-dir', 'not-given')
```

Comment:
Does Meson allow definition of functions? If so, perhaps could be written as:

```
incdir_numpy = meson.get_external_property('numpy-include-dir', installed_numpy_include_dir())
```

Reply:
(But this is totally fine the way it is too)

Reply:
I don't believe this can be accomplished in Meson. Even if it were possible, it might also have the unintended side effect of always invoking Python on the build system to try to determine the include paths, even when users specify custom locations as properties.